### PR TITLE
FLUO-950 Make VisibilityCache customizable

### PR DESCRIPTION
**Pull request by kpm1985**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/fluo/core/impl/VisibilityCache.java [hunk 1]
```diff
@@ -48,9 +48,9 @@ public int weigh(Bytes key, ColumnVisibility vis) {
 
   VisibilityCache() {
```
Comment (keith-turner) [inline modules/core/src/main/java/org/apache/fluo/core/impl/VisibilityCache.java]: You can modify this constructor to take a `FluoConfiguration`.  The viscache constructor is only called in `SharedResources` which can be changed to do `visCache = new VisibilityCache(env.getConfiguration())`

Comment (kpm1985) [reply modules/core/src/main/java/org/apache/fluo/core/impl/VisibilityCache.java]: Ok @keith-turner this should all be taken care of with this next commit

**Comment by kpm1985** [comment]
@keith-turner
I've pushed some changes. Build fails but I looked it seems unrelated
(timeout stuff again).

File: modules/core/src/main/java/org/apache/fluo/core/impl/FluoConfigurationImpl.java [hunk 2]
```diff
@@ -143,6 +143,45 @@ public static long getTxIfoCacheTimeout(FluoConfiguration conf, TimeUnit tu) {
     }
     return tu.convert(millis, TimeUnit.MILLISECONDS);
   }
+  public static final String VISIBILITY_CACHE_SIZE = FLUO_IMPL_PREFIX + ".visibility.cache.size.mb";
+  public static final long VISIBILITY_CACHE_SIZE_DEFAULT = 10000000;
```
Comment (keith-turner) [inline modules/core/src/main/java/org/apache/fluo/core/impl/FluoConfigurationImpl.java]: I think in java 7 or 8 the ability to add underscores in numbers was added.  So could change this to `10_000_000` to make it a bit more readable.

File: modules/core/src/main/java/org/apache/fluo/core/impl/VisibilityCache.java [hunk 3]
```diff
@@ -20,14 +20,16 @@
...
-import com.google.common.cache.Cache;
-import com.google.common.cache.CacheBuilder;
-import com.google.common.cache.Weigher;
 import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.fluo.api.config.FluoConfiguration;
 import org.apache.fluo.api.data.Bytes;
 import org.apache.fluo.api.data.Column;
 import org.apache.fluo.core.util.ByteUtil;
 
+import com.google.common.cache.Cache;
```
Comment (keith-turner) [inline modules/core/src/main/java/org/apache/fluo/core/impl/VisibilityCache.java]: When I ran `mvn compile` locally it reorganized these imports.

**Comment by kpm1985** [comment]
Yes, importing FluoConfiguration with shift O in eclipse did that, forgot
to run impsort and commit again. That plugin is so boss I've been
recommending it on Log4J2 and Nutch.

File: modules/core/src/main/java/org/apache/fluo/core/impl/VisibilityCache.java [hunk 4]
```diff
@@ -46,11 +47,11 @@ public int weigh(Bytes key, ColumnVisibility vis) {
 
   private final Cache<Bytes, ColumnVisibility> visCache;
 
-  VisibilityCache() {
+  VisibilityCache(FluoConfiguration conf) {
     visCache = CacheBuilder.newBuilder()
-        .expireAfterAccess(FluoConfigurationImpl.TX_INFO_CACHE_TIMEOUT_DEFAULT,
+        .expireAfterAccess(FluoConfigurationImpl.getTxIfoCacheTimeout(conf, TimeUnit.MILLISECONDS),
```
Comment (keith-turner) [inline modules/core/src/main/java/org/apache/fluo/core/impl/VisibilityCache.java]: This should call `getVisibilityCacheTimeout()`

**Comment by kpm1985** [comment]
@keith-turner Nice catch!

**Comment by kpm1985** [comment]
@keith-turner Let me know if this looks ok to you, and I'll start on FLUO-939 next.

**Comment by keith-turner** [comment]
@kpm1985 looks good, I squashed and pushed commit 9e87f47

**Pull request closed without merging**